ATI mini 45 FT sensor setup

Kotochleb opened the pull request.

jmirabel commented on robots/fer/fer.urdf.xacro:
```diff
@@ -68,4 +72,17 @@
   </xacro:franka_robot>
-</robot>
+
+  <!-- ATI sensors have sampling rate as 7000 divided by an integer -->
+  <!-- Dividing by 3 is the closest to 2 times 1000 Hz which is the -->
+  <!-- control frequency of Panda -->
```
Why do we want a frequency of 2kHz and not 1kHz ?
I though we controlled the Panda at 1kHz.

Kotochleb replied:
I had a discussion with a friend of mine who was using those sensors and he suggested to me to go 2x the control frequency. We can go down to 1 kHz if you want

jmirabel replied:
Naively, I would assume that, when the frequencies do not match, what matters is the maximal delay.

Since the sensor if the fastest, the maximal delay is the period of the sensor measurement. At 7e3 / 3 ~= 2333Hz, the maximal delay would be about 429ms.

- Do we have any reason not to have the sensor give values at 7kHz (i.e. max delay of 143ms) ?

Kotochleb replied:
I didn't want to go with full 7 kHz as I was worried that we might end up having throughput issues or inconsistent packet handling at the kernel level. My way of thinking is to first get the calibration setup working and then once we will want to increase precision, we might play with frequencies of measurement and maybe internal filtering. Tho I didn't pay much attention to reading the datasheet of the sensor.

jmirabel replied:
I think it is rather the opposite. We need the maximal precision for calibration. If it works, I would simply use 7kHz for calibration and something slower for other usages.
Calibration requires to perform dynamical motions with a good synchronization between the sensors. While at runtime, it is less important that we have a good synchronization, especially if we have a good calibration.

jmirabel commented on end_effectors/ft_sensors/inertials.yaml:
```diff
@@ -0,0 +1,12 @@
+    zz: 0.000107824
+    xy: 0
+    xz: 0
+    yz: 0
```
How did you come up with those values?

Kotochleb replied:
I should've made a comment on this. Mass is measured with a scale, while inertia is computer based on that mass and an STL file

Kotochleb replied:
Comment added

jmirabel commented on end_effectors/ft_sensors/ati_mini45.urdf.xacro:
```diff
@@ -0,0 +1,78 @@
+    <xacro:property name="length" value="0.0327" />
+    <xacro:property name="ee_inertials" value="${sensor_inertials}" />
+
+    <joint name="${prefix}hand_joint" type="fixed">
```
I would prefer a name that is specific to the sensor itself.
```suggestion
    <joint name="${prefix}ati_mini45_base" type="fixed">
```
or `ati_mini45_base_joint` or just `ati_mini45_joint`.

Kotochleb replied:
Fixed

Kotochleb commented:
Since the last review, I moved things around, hopefully keeping backwards compatibility with code with no force sensor at all. For the mass calibration, I decided not to use it in here, as there are in total 3 elements (sensor, camera and hand itself) that have separately defined intervals. Additionally, I added support for Gazebo (jointly with https://github.com/agimus-project/franka_ros2/pull/5). For now, I did not manage to get the sensor to work properly in the simulation tho. ros2_control sees the interface and generally for now I think this is enough, tho values are all the time zeros. After googling a bit and copying sdf definition from unit tests of ign-gz 6 I still didn't manage to have it working. `ign gazebo --topic --list` does not show the topic, even though it is visible as registered in the UI of Gazebo. Anyway. For now, I think this is good enough.

MaximilienNaveau commented:
This a bug of ros2_control_gazebo on humble you won't be able to use this as is...
This quite annoying and we may want to move to Jazzy if need be.

Kotochleb commented:
It is not a  bug. It was a missing feature which I added here https://github.com/agimus-project/franka_ros2/pull/5
The bug might only be on the pure Gazebo side in this case, but the code seems to be there
